### PR TITLE
Feature/authentication readme

### PR DESCRIPTION
**Pull request by AndreaLandiArk**
Added Authentication instruction for Authentication Provider Module

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -93,23 +117,134 @@ externalFilters?:  boolean;
...
+
 **useQueryHook:** The query hook generated by your Redux Toolkit Query API for fetching table data that should return an object like this
 
-    data: {
-    	data:  retData.data,
-    	count:  retData.count,
-    	page:  page,
-    	limit:  pageSize,
-    }
+data: {
```
Comment (AndreaCuneo) [inline README.md:127]: this should be in a code section

File: README.md [hunk 2]
```diff
@@ -93,23 +117,134 @@ externalFilters?:  boolean;
...
+
+This project is implemented with a flexible authentication provider that can support multiple providers.
+Now it support **MSAL** and **AUTH0** providers and you can switch between one another easily.
+This is how:
+
+1.  Go to the **index.tsx** file
+2.  Instantiate the implementation of your choice of AuthProvider interface
+
+```Typescript
+const  authProvider  =  new  Auth0AuthProvider(env);
```
Comment (AndreaCuneo) [inline README.md:157]: what is env? where is created from?

File: README.md [hunk 3]
```diff
@@ -93,23 +117,134 @@ externalFilters?:  boolean;
...
+
+in this case we choose the **Auth0** implementation.
+_env_ is the enviroment and it must contains all the data needed to authenticate with the provider.
+For this reason make sure that the **connectionStrings.js** file is aligned with the deploy environments you are using.
+In order to make this works locally you must create a **.env.local** file in the root of your project with all the env variables needed by connectionStrings.js and your AuthProvider implementation.
+example for **Auth0** :
+
+### .env.local
+
+```javascript
```
Comment (AndreaCuneo) [inline README.md:168]: this shuold go in a section dedicated on how to configure Auth0.
the same is needed for MSAL

**Review by AndreaCuneo — CHANGES_REQUESTED** [review]
Also missing:
- authenticated only routes
- authenticated only sections (how to hide a component if user is not logged? please provide a library component <AuthOnly fallback={null | something} > <stuff to hide > </AuthOnly>

File: README.md (hunk 3 above)
Comment (AndreaLandiArk) [reply README.md:168]: this wanted to be just an example, or we have to implement the readme for each authentication provide we may need and also update them when a change occour

File: README.md (hunk 2 above)
Comment (AndreaLandiArk) [reply README.md:157]: it's explained 2 rows below

File: README.md (hunk 3 above)
Comment (AndreaCuneo) [reply README.md:165]: either the Providers use the config object from the 3rd party library or, like seems you've done, you define a different Config to be passed to the ctor of the Provider, then yes documentation is needed.

File: README.md (hunk 2 above)
Comment (AndreaCuneo) [reply README.md:154]: it's described how to configure it locally, not what shall contain to work.

File: src/lib/authentication/authenticationComponents.tsx [hunk 4]
```diff
@@ -8,5 +8,6 @@ export function AuthenticatedOnly({
   component: () => ReactNode;
```
Comment (AndreaCuneo) [inline src/lib/authentication/authenticationComponents.tsx:8]: use children

**Review by AndreaCuneo — CHANGES_REQUESTED** [review]

**Review by AndreaCuneo — APPROVED** [review]

**Review by sean-sligo — APPROVED** [review]

**Pull request merged**